### PR TITLE
Add support for linting example scripts

### PR DESCRIPTION
**Pull request by gkjohnson**
I feel like I've see quite a few PRs that introduce code-style issues into the examples files -- this should make it easier to automatically catch those problems. This is can be changed to lint the module files (or both?) once #16920 is merged.

_Note_ this is expected to fail until the linting issues are fixed.

In another PR we can enable linting on the HTML files

### CODE REVIEW

**Comment by Mugen87** [comment]
> Note this is expected to fail until the linting issues are fixed.

Maybe it's better to first fix the styles before merging this. Otherwise travis will die for all PRs in the next time until somebody cleans everything up. I'd like to avoid that since everyone will ignore travis and miss real/important linter/test errors.

**Comment by gkjohnson** [comment]
> Maybe it's better to first fix the styles before merging this.

Oh I definitely agree. I added this now just to propose the idea (and see how many linting errors there actually were).

I'd start fixing the issues now but with #16920 it seems unclear whether or not it's worth proposing new fixes to the js example files or waiting to lint the jsm ones. Any thoughts?

**Comment by Mugen87** [comment]
I think we can wait. The issue is not that urgent. I would proceed with this PR as soon as #16920 lands.

**Comment by Mugen87** [comment]
Okay, since #16920 will not be merged, it's okay to solve the linter issues for all js files and the run `node/modularize.js`.

It seems that almost all linter errors are produce in the `js/shaders` directory. Additional tabs are used to format the GLSL shader code. The problem is instead of this:
```
"void main() {",

    "gl_FragColor = vec4( 1.0, 0.0, 0.0, 0.5 );",

"}"
``` 
we would have to do this:
```
"void main() {",

"    gl_FragColor = vec4( 1.0, 0.0, 0.0, 0.5 );",

"}"
``` 
So the padding is applied within the string. I'm not sure somebody want's to do this by hand...

**Comment by gkjohnson** [comment]
@Mugen87 Sounds good to me -- I'll update this PR when I get the chance and can start contributing linting fixes to the examples.

**Comment by gkjohnson** [comment]
Just updated the examples lint command -- once #17542 and #17541 are merged this should pass the CI check.

File: package.json [hunk 1]
```diff
@@ -52,7 +52,9 @@
     "dev": "concurrently --names \"ROLLUP,HTTP\" -c \"bgBlue.bold,bgGreen.bold\" \"rollup -c utils/build/rollup.config.js -w -m inline\" \"http-server -c-1 -p 8080\"",
     "dev-test": "concurrently --names \"ROLLUP,ROLLUPTEST,HTTP\" -c \"bgBlue.bold,bgRed.bold,bgGreen.bold\" \"rollup -c utils/build/rollup.config.js -w -m inline\" \"rollup -c test/rollup.unit.config.js -w -m inline\" \"http-server -p 8080\"",
     "start": "npm run dev",
-    "lint": "eslint src --ext js --ext ts && tsc -p utils/build/tsconfig.lint.json",
+    "lint": "npm run lint-source && npm run lint-examples",
```
Comment (Mugen87) [inline package.json]: How about writing the commands like so:
```
"lint": "npm run lint-source && npm run lint-examples && npm run lint-typescript",
"lint-source": "eslint src --ext js --ext ts",
"lint-examples": "eslint examples/jsm --ext js --ext ts --ignore-pattern libs",
"lint-typescript": "tsc -p utils/build/tsconfig.lint.json"
```
IMO, the commands are more clear in this way.

Comment (gkjohnson) [reply package.json]: I agree! Just updated

**Comment by gkjohnson** [comment]
Okay this should pass now -- it looks like there are still some linter warnings mostly of the "no-unused-vars" flavor but those shouldn't cause CI to break and can be fixed in subsequent PRs.

**Review by Mugen87 — APPROVED** [review]

**Comment by Mugen87** [comment]
I remember now that @mrdoob does not want to apply strict linter rules to the examples code in order to simplify contribution. I think we can get this merged if we let the current lint command unchanged (`npm run lint`) and just add `npm run lint-examples`.

**Comment by gkjohnson** [comment]
@Mugen87 Sounds good -- I've updated the PR so there are just `lint` and `lint-examples` scripts.

The `lint` script lints the js and ts files in /src and the `lint-examples` script lints the js and ts files in /examples/jsm!

**Comment by mrdoob** [comment]
LGTM!

**Comment by Mugen87** [comment]
> The lint script lints the js and ts files in /src

It actually lints the TS files in `examples`, too. I guess we should change `tsconfig.lint.json` so it works like you have described.

**Comment by gkjohnson** [comment]
> It actually lints the TS files in examples, too. I guess we should change tsconfig.lint.json so it works like you have described.

Yup! Sorry looks like I forgot to commit that change. I've added it now!

**Comment by Mugen87** [comment]
@mrdoob This looks good now!

**Comment by Mugen87** [comment]
Let's merge this 👍

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**